ipxe build failures on 19.03

## Issue description

We use ipxe based on clever's setup (https://github.com/cleverca22/nixos-configs/blob/master/netboot_server.nix) like this:

```
  ipxe' = pkgs.ipxe.overrideDerivation (drv: {
    installPhase = ''
      ${drv.installPhase}
      make $makeFlags bin-x86_64-efi/ipxe.efi bin-i386-efi/ipxe.efi
      cp -v bin-x86_64-efi/ipxe.efi $out/x86_64-ipxe.efi
    '';
  });
```

This causes the following build failure on 19.03:
```
  [HOSTCC] util/elf2efi64
  [VERSION] bin-x86_64-efi/version.ipxe.efi.o
  [LD] bin-x86_64-efi/ipxe.efi.tmp
  [FINISH] bin-x86_64-efi/ipxe.efi
Unrecognised relocation type 4
make[1]: *** [Makefile.efi:26: bin-x86_64-efi/ipxe.efi] Error 1
rm bin-x86_64-efi/version.ipxe.efi.o
make: *** [Makefile.housekeeping:259: bin-x86_64-efi/ipxe.efi] Error 2
builder for '/nix/store/y1nv3mpmmfm0wmxhwdxazdmf7g2l5wfh-ipxe-20180220-47849be.drv' failed with exit code 2
cannot build derivation '/nix/store/x28j6ym3dj9jy217wrg0vln93xr0v6xa-morph.drv': 1 dependencies couldn't be built
error: build of '/nix/store/x28j6ym3dj9jy217wrg0vln93xr0v6xa-morph.drv' failed
Error while running `nix build ...`: See above.ERROR: Job failed: exit status 1
```

Googling around for "Unrecognised relocation type 4" and ipxe returns https://bugzilla.opensuse.org/show_bug.cgi?id=1105510, indicating it's fixed upstream (it is, I have a PR coming).


### Steps to reproduce

See the override above

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.1, NixOS, 19.03.git.5f3be9b (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `""`
 - channels(atu): `""`
 - nixpkgs: `/etc/nixos/nixpkgs`

--- a/pkgs/tools/misc/ipxe/default.nix
+++ b/pkgs/tools/misc/ipxe/default.nix
@@ -3,8 +3,8 @@
 }:
 
 let
-  date = "20180220";
-  rev = "47849be3a900c546cf92066849be0806f4e611d9";
+  date = "20190310";
+  rev = "b6ffe28a21c53a0946d95751c905d9e0b6c3b630";
 in
 
 stdenv.mkDerivation {
@@ -14,7 +14,7 @@ stdenv.mkDerivation {
 
   src = fetchgit {
     url = git://git.ipxe.org/ipxe.git;
-    sha256 = "1f4pi1dp2zqnrbfnggnzycfvrxv0bqgw73dxbyy3hfy4mhdj6z45";
+    sha256 = "1s8sy75lpx8zq60wc0i35d8c1wwm1rq3scxpkq31623d097mch59";
     inherit rev;
   };
 

